### PR TITLE
Abstract post body and get parameters parsing

### PR DESCRIPTION
**Pull request by ioigoume**
#101

### CODE REVIEW

File: src/Controller/Traits/RequestTrait.php [hunk 1]
```diff
@@ -36,18 +37,22 @@ trait RequestTrait
      */
     protected string $expectedStateAuthId = OAuth2::AUTHID;
 
+    /** @var array|null  */
+    private ?array $requestParams;
```
Comment (pradtke) [inline src/Controller/Traits/RequestTrait.php:41]: I don't see anything that sets this object field. Only something that reads it.

File: src/Lib/RequestUtilities.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+use Symfony\Component\HttpFoundation\Request;
+
+class RequestUtilities
+{
+    /**
+     * @param   Request  $request
+     *
+     * @return Generator
+     */
+    public static function parseRequestGenerator(Request $request): Generator
```
Comment (pradtke) [inline src/Lib/RequestUtilities.php:17]: Is this method used anywhere?

**Review by pradtke — CHANGES_REQUESTED** [review]
I did some tests and I think we also will need some adjustments in `OAuth2Controller->linkback()` to handle the response being a POST.  Currently the code assumes the `code` is a query parameter.

File: src/Lib/RequestUtilities.php (hunk 2 above)
Comment (ioigoume) [reply src/Lib/RequestUtilities.php]: Removed it.

File: src/Controller/Traits/RequestTrait.php (hunk 1 above)
Comment (ioigoume) [reply src/Controller/Traits/RequestTrait.php]: I updated the code and it is now used as it should be.

**Comment by ioigoume** [comment]
> I did some tests and I think we also will need some adjustments in `OAuth2Controller->linkback()` to handle the response being a POST. Currently the code assumes the `code` is a query parameter.

Hi @pradtke , i refactored the code according to your comments.

**Review by pradtke — APPROVED** [review]

**Comment by pradtke** [comment]
Thank you @ioigoume .  Apple logins are working for me again.

**Pull request merged**